### DISCUSSION TITLE
Reaction measurement in multilayered soil

### DISCUSSION THREAD

**Discussion by Francisliii** [Anura3D/Anura3D_OpenSource#254] [Q&A]
Hi Dear community,

I am simulating the cone penetration process in the multilayered soil profile. My model in the homogeneous soil profile runs smoothly and the reaction profile looks reasonable. When it comes to the multilayered soil profile, although the penetration process looks no problem, the simulated reaction values at the tip always reduces to zero passing the interface of the first and underlaying soil layer. The soil strength parameter values across the interface are not so different so the reaction profile is definitely incorrect.

Please see below.
![1724482296271](https://github.com/user-attachments/assets/40c6e220-4cf0-41bd-b356-8ef84d1f03cb)

My gut feeling is that the contact property was not properly applied in the code entering the second soil layer. I am trying to debug this issue but just wondering if anyone in the community knows any trick to solve this problem.

Many thanks,
Francis

**Comment by HayThem-JDN** [comment]
I had the same experience. When modelling a typical footing problem on layered soil. it came to my attention that the final reaction force should be the sum of the Y component of all material ID (except the footing of course)

**Comment by HayThem-JDN** [comment]
Which approach you were using in your simulation? load (traction A) or velocity approach?

thanks in advance

**Reply by Francisliii** [reply]
Velocity approach
